server/services/authorization/canAccessRoom.ts

```diff
@@ -85,10 +106,11 @@ export const canAccessRoom: RoomAccessValidator = async (room, user, extraData):
 	// if (!room || !user) {
 	// 	return false;
 	// }
-
 	for await (const roomAccessValidator of roomAccessValidators) {
 		if (await roomAccessValidator(room, user, extraData)) {
-			return true;
+			if (await roomViewPermission(room.t, user._id)) {
```

Comment:
```suggestion
			if (await hasRoomViewPermission(room.t, user._id)) {
```
If we decide to update the function's name, then we'd have to update it here too